src/test/kotlin/no/nav/familie/klage/behandling/enhet/BehandlendeEnhetServiceTest.kt

```diff
@@ -0,0 +1,157 @@
+package no.nav.familie.klage.behandling.enhet
+
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import no.nav.familie.klage.behandling.BehandlingService
+import no.nav.familie.klage.behandling.dto.OppdaterBehandlendeEnhetRequest
+import no.nav.familie.klage.behandlingshistorikk.BehandlingshistorikkService
+import no.nav.familie.klage.behandlingshistorikk.domain.HistorikkHendelse
+import no.nav.familie.klage.fagsak.FagsakService
+import no.nav.familie.klage.infrastruktur.exception.Feil
+import no.nav.familie.klage.oppgave.OppgaveService
+import no.nav.familie.klage.testutil.DomainUtil.behandling
+import no.nav.familie.klage.testutil.DomainUtil.fagsak
+import no.nav.familie.kontrakter.felles.klage.Fagsystem
+import no.nav.familie.kontrakter.felles.klage.Stønadstype
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.EnumSource
+
+class BehandlendeEnhetServiceTest {
+    private val behandlingService: BehandlingService = mockk()
+    private val fagsakService: FagsakService = mockk()
+    private val behandlingshistorikkService: BehandlingshistorikkService = mockk()
+    private val oppgaveService: OppgaveService = mockk()
+    private val behandlendeEnhetService =
+        BehandlendeEnhetService(
+            behandlingService = behandlingService,
+            fagsakService = fagsakService,
+            behandlingshistorikkService = behandlingshistorikkService,
+            oppgaveService = oppgaveService,
+        )
+
+    @Nested
+    inner class OppdaterBehandlendeEnhet {
+        @ParameterizedTest
+        @EnumSource(BarnetrygdEnhet::class, names = ["MIDLERTIDIG_ENHET"], mode = EnumSource.Mode.EXCLUDE)
+        fun `skal oppdatere behandlende enhet på behandling og oppgave samt opprette historikkinnslag`(
```

Comment:
Kanskje greit å skille på disse to? 

```suggestion
        fun `skal oppdatere behandlende enhet på behandling og oppgave samt opprette historikkinnslag for BA`(
```